receiver/vcenterreceiver/scraper.go

```diff
@@ -0,0 +1,373 @@
+// Copyright  The OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package vcenterreceiver // import "github.com/open-telemetry/opentelemetry-collector-contrib/receiver/vcenterreceiver"
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"strings"
+	"time"
+
+	"github.com/vmware/govmomi/object"
+	"github.com/vmware/govmomi/vim25/mo"
+	"github.com/vmware/govmomi/vsan/types"
+	"go.opentelemetry.io/collector/component"
+	"go.opentelemetry.io/collector/model/pdata"
+	"go.opentelemetry.io/collector/receiver/scrapererror"
+	"go.uber.org/zap"
+
+	"github.com/open-telemetry/opentelemetry-collector-contrib/receiver/vcenterreceiver/internal/metadata"
+)
+
+// example 2022-03-10 14:15:00
+const timeFormat = "2006-01-02 15:04:05"
+
+var _ component.Receiver = (*vcenterMetricScraper)(nil)
+
+type vcenterMetricScraper struct {
+	client      *vcenterClient
+	mb          *metadata.MetricsBuilder
+	logger      *zap.Logger
+	vsanEnabled bool
+}
+
+func newVmwareVcenterScraper(
+	logger *zap.Logger,
+	config *Config,
+) *vcenterMetricScraper {
+	client := newVmwarevcenterClient(config)
+	return &vcenterMetricScraper{
+		client:      client,
+		logger:      logger,
+		mb:          metadata.NewMetricsBuilder(config.MetricsConfig.Metrics),
+		vsanEnabled: true,
+	}
+}
+
+func (v *vcenterMetricScraper) Start(ctx context.Context, _ component.Host) error {
+	return v.client.Connect(ctx)
+}
+
+func (v *vcenterMetricScraper) Shutdown(ctx context.Context) error {
+	return v.client.Disconnect(ctx)
+}
+
+func (v *vcenterMetricScraper) scrape(ctx context.Context) (pdata.Metrics, error) {
+	if v.client == nil {
+		return pdata.NewMetrics(), errors.New("no SDK client instantiated")
+	}
+
+	// ensure connection before scraping
+	if err := v.client.Connect(ctx); err != nil {
```

Comment:
Does it need to connect every time?

Reply:
This I thought was based off this previous comment https://github.com/open-telemetry/opentelemetry-collector-contrib/pull/9224#discussion_r849812425

I thought that if there is a chance after a network outage that the scrape would work we should try to connect? Or are you suggesting we try to keep track of broken connection starts here rather than letting govmomi do that for us?

Reply:
I'm not sure what you are saying here. 

We need to be tolerant of temporary connection outages, but that does not mean we need to instruct the client to connect every time. I see that internally the func does not actually try to connect every time, so perhaps it is just a misnomer here.

Reply:
Yeah I'll go ahead and rename the func to `EnsureConnection` hopefully to resolve any confusion!